test/processors/feedback_url_swapper_test.rb

```diff
@@ -0,0 +1,87 @@
+require "test_helper"
+
+class FeedbackURLSwapperTest < MiniTest::Test
+  def test_should_replace_input_url_values_in_gem_layout
+    template = as_nokogiri %(
+      <html>
+        <body>
+          <div class="gem-c-feedback">
+            <input
+              id="test-input"
+              type="hidden"
+              name="email_survey_signup[survey_source]"
+              value="/old_path"
+            >
+            <input
+              id="test-input-two"
+              type="hidden"
+              name="url"
+              value="https://example.com/old_path"
+            >
+          </div>
+        </body>
+      </html>
+    )
+
+    request = {}
+
+    def request.base_url
+      "https://new-example.com"
+    end
+
+    def request.fullpath
+      "/new_path"
+    end
```

Comment:
We can use rack to create a request object for testing:

```
    env = Rack::MockRequest.env_for("https://new-example.com/new_path")
    request = Rack::Request.new(env)
```

Reply:
I wish I'd known about this before - thanks for this!